packages/keyring-controller/src/KeyringController.ts

```diff
@@ -335,11 +336,28 @@ type SessionState = {
   password?: string;
 };
 
+export type EncryptionResultConstraint<SupportedKeyMetadata> = {
+  salt?: string;
+  keyMetadata?: SupportedKeyMetadata;
+};
+
+export type DefaultEncryptionResult<SupportedKeyMetadata> = {
+  data: string;
+  iv: string;
+  salt?: string;
+  keyMetadata?: SupportedKeyMetadata;
+};
```

Comment:
I'm a bit confused about the different EncryptionResult types here:
Why don't we have a unified `EncryptionResult` interface?
Can you give some context?

Reply:
Extension (or `browser-passworder`) uses `data` as ciphertext property, while mobile uses `cipher`. The `EncryptionResultConstraint` represent the minimum object to be extended as encryption result type, while `DefaultEncryptionResult` represents the type coming from `browser-passworder` (which is also the default type parameter, hence its name).

Splitting them allows us to make the `Encryptor` type generic enough, `KeyringController` more type safe as it can always access `keyMetadata` and `salt`, and allows Mobile/Extension to define their EncryptionResult type